### PR TITLE
#160673841 Restrict editing comment to the owner only

### PR DESCRIPTION
**Pull request by hariclerry**
#### What does this PR do?
This PR adds a condition to ensure that posted comments are edited by their owners only.

#### Description of Task to be completed?
- Fix the user comment editing right

#### How should this be manually tested?
- Clone the repository
- Run `make start` docker build command on the terminal to install all dependencies and run the App in the docker development environment server.
- On the browser, type the localhost URL ` http://localhost:3000` and Login with your Andela email account into the App.
- On the Requests page, click on the request Id from the list of requests displayed on the request table, you should be able to see the Request details Modal displayed.
- Add a new comment using the comment box.
- On the displayed added comment, click on the `edit` button to edit the comment.
- When done with editing and saving the changes, you should be able to see the edited comment displayed and labeled `edited`.

#### Any background context you want to provide?
To test that users can edit the comments belonging to them only, there must be two different users logged in, one as a `Requester` and the other user as a `Manager` respectively.

#### What are the relevant pivotal tracker stories?
[#160673841](https://www.pivotaltracker.com/story/show/160673841)

#### Screenshots (if appropriate)
Screenshot showing comments by different users with editing option.

screen shot 2018-09-21 at 03 44 24

#### Questions:
None

### CODE REVIEW

File: src/components/RequestsModal/UserComments/UserComments.jsx [hunk 1]
```diff
@@ -73,13 +71,13 @@ class UserComments extends Component {
...
-                onClick={() => this.editComment(comment)}>
+            {email === comment.userEmail ? (
+              <span className="modal__dialog">
+                <button 
+                  type="button" className={`edit-button ${activeCommentId === comment.id ? 'active': ''}`} onClick={() => this.editComment(comment)}>
                 Edit
-              </button>
-            </span>
+                </button>
+              </span>) : null }
```
Comment (samuelkubai) [inline src/components/RequestsModal/UserComments/UserComments.jsx]: Please prevent the delete button too.

Comment (hariclerry) [reply src/components/RequestsModal/UserComments/UserComments.jsx]: @samuelkubai, done!

**Comment by samuelkubai** [comment]
@hariclerry please fix the failing tests? You must have tweaked something with the new commit.

**Comment by samuelkubai** [comment]
@hariclerry Awesome work.

**Pull request merged**